rotkehlchen/chain/evm/decoding/thegraph/balances.py

```diff
@@ -40,6 +46,78 @@ def __init__(
         self.token = native_asset.resolve_to_evm_token()
         self.staking_contract = staking_contract
 
+    def _query_vesting_balances(self, balances: 'BalancesSheetType') -> 'BalancesSheetType':
+        """Query balances of vested GRT tokens if DelegationTransfer events are found.
+        This function modifies and returns the 'balances'"""
+        db_filter = EvmEventFilterQuery.make(
+            counterparties=[self.counterparty],
+            location=Location.ETHEREUM,
+            event_types=[HistoryEventType.INFORMATIONAL],
+            event_subtypes=[HistoryEventSubType.NONE],
+        )
+        with self.event_db.db.conn.read_ctx() as cursor:
+            events = self.event_db.get_history_events(
+                cursor=cursor,
+                filter_query=db_filter,
+                has_premium=True,
+            )
+        if len(events) == 0:
+            return balances
+
+        vesting_contracts = defaultdict(list)
+        for event in events:
+            if (
+                event.location_label is None or
+                event.extra_data is None or
+                (delegator := event.extra_data.get('delegator_l2')) is None
+            ):
+                continue
+            vesting_contracts[string_to_evm_address(event.location_label)].append(
+                EvmContract(
+                    address=string_to_evm_address(delegator),
+                    abi=GRAPH_TOKEN_LOCK_WALLET_ABI,
+                    deployed_block=0,
+                ),
+            )
+
+        log.info(f'{vesting_contracts=}')
+        if len(vesting_contracts) == 0:
+            return balances
+
+        grt_price = Inquirer.find_usd_price(self.token)
+        calls: list[tuple[ChecksumEvmAddress, str]] = []
+        call_mapping: list[tuple[ChecksumEvmAddress, EvmContract]] = []
+        for contract_address, delegators in vesting_contracts.items():
+            for delegator in delegators:
+                encoded_call = delegator.encode(method_name='totalOutstandingAmount')
+                calls.append((delegator.address, encoded_call))
+                call_mapping.append((contract_address, delegator))
```

Comment:
can't you do this too when you create the contract in the previous loop? that prevents you from iterating again?

so you create the contract (because you need it later but you also do this encode that you have here

Reply:
let me know if it doesn't make sense what I'm saying